xgcm/test/test_metrics.py

```diff
@@ -96,6 +97,38 @@ def test_assign_metric(key, metric_vars):
     _ = Grid(ds, coords=coords, metrics={key: metric_vars})
 
 
+@pytest.mark.parametrize(
+    "axes, expected_axes",
+    [
+        (
+            ("X", "Y"),
+            (
+                (frozenset({"X", "Y"}),),
+                (frozenset({"X"}), frozenset({"Y"})),
+                (frozenset({"Y"}), frozenset({"X"})),
+            ),
+        ),
+        (
+            ("X", "Y", "Z"),
+            (
```

Comment:
I'm having difficulty with parameterizing this test, because when I use three axes `("X","Y","Z")` each time I run `iterate_axis_combinations` the output can be slightly different. For example:
```python
(frozenset({"Y","X"}), frozenset({"Z}))
```
can become:
```python
(frozenset({"X","Y"}), frozenset({"Z}))
```
It becomes difficult to set `expected_axes` because I can't really predict the exact output from `iterate_axis_combinations`. I might have to rethink how the test is being done.

Reply:
```
a = {
    (frozenset({"Y", "X"}), frozenset({"Z"})), (frozenset({"X", "Z"}), frozenset({"Y"}))
}
b = {
    (frozenset({"X", "Y"}), frozenset({"Z"})), (frozenset({"Z", "X"}), frozenset({"Y"}))
}

a == b
```

is `True` so I'm confused about what you're running in to.

Reply:
When I use
```python
assert iterate_axes == expected_axes
```
I get an `AssertionError: Use -v to get the full diff`, so I tried using
```python
set_diff = iterate_axes.difference(expected_axes)
assert set_diff == set()
```
which works for `axis = ("X","Y")` but not for `axis = ("X","Y","Z")`, and I noticed that the order of the frozensets would change sometimes so I thought that might be it, but maybe something else is causing the error.

Reply:
I tried adding these elements to `expected_axes` and the test passed. I think I was too preoccupied with the debug function and trying to get the sets to match exactly each time. X_________x